libvast/include/vast/type.hpp

```diff
@@ -346,6 +346,14 @@ class type final : public stateful_type_base {
   /// Returns all aliases of this type, excluding this type itself.
   [[nodiscard]] detail::generator<type> aliases() const noexcept;
 
+  /// Resolves an extractor on this type.
+  [[nodiscard]] detail::generator<offset>
+  resolve(std::string_view extractor) const noexcept;
+
+  /// Resolves an extractor on this type.
+  [[nodiscard]] detail::generator<offset>
+  resolve(std::vector<std::string_view> extractor) const noexcept;
```

Comment:
Just curious from DX perspective because it's not (yet) documented: what's the difference in semantics over the singleton overload? Is the same as calling N times `resolve`, just faster?

Reply:
Yes. I identified at some point during development that for transform steps especially we often resolved a bunch of extractors, sorted the results and then de-duplicated them. We also get this for free as the algorithm works on a vector of string views anyways.

This will become obvious once I've got the remaining bits and pieces in. I'm just extracting what I've got piece-by-piece while making finishing touches like writing documentation.